rpc/handlers.go

```diff
@@ -1031,6 +1033,13 @@ func (h *Handler) TransactionStatus(hash felt.Felt) (*TransactionStatus, *jsonrp
 	case ErrTxnHashNotFound:
 		txStatus, err := h.feederClient.Transaction(context.Background(), &hash)
 		if err != nil {
+			// Check if the error is due to a transaction not being found
+			notFoundErr := fmt.Sprintf("%d %s", http.StatusNotFound, http.StatusText(http.StatusNotFound))
```

Comment:
I get an `http.StatusOK` when requesting a tx that doesn't exist, so I don't think this will work. 

I get the following response on testnet ([example](https://alpha4.starknet.io/feeder_gateway/get_transaction?transactionHash=0x2926a3b3c8f1cf08b1597e153928fc49bfaa2c521952610c108808fc21fe285)), which is on v0.12.1:

```
{"finality_status": "NOT_RECEIVED", "status": "NOT_RECEIVED"}
```

and this response on [mainnet](https://alpha-mainnet.starknet.io/feeder_gateway/get_transaction?transactionHash=0x2926a3b3c8f1cf08b1597e153928fc49bfaa2c521952610c108808fc21fe285):

```
{"status": "NOT_RECEIVED"}
```
I think we just need to add this `NOT_RECEIVED` enum value to `feeder.TxnExecutionStatus` (not sure if that's the exact name).

Reply:
Let's add tests for these cases as well. See `handlers_test.go`.